### PR TITLE
[TVM EP][CI] Added TVMso EP testing into CI

### PR DESCRIPTION
**Pull request by KJlaccHoeUM9l**
In one of our previous [PR#11389](https://github.com/microsoft/onnxruntime/pull/11389) we added the ability to use precompiled `.so` libraries.
In this PR, we add testing for this functionality into public ORT CI.

Also in this PR, small refactor was carried out for test of model with dynamic shapes.

### CODE REVIEW

**Comment by KJlaccHoeUM9l** [comment]
Hello @xadupre!
Could you review this PR?

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -1973,13 +1973,11 @@ def nuphar_run_python_tests(build_dir, configs):
...
 def tvm_run_python_tests(build_dir, configs):
     for config in configs:
-        if config == "Debug":
-            continue
         cwd = get_config_build_dir(build_dir, config)
         if is_windows():
             cwd = os.path.join(cwd, config)
-        dll_path = os.path.join(build_dir, config, "_deps", "tvm-build", config)
-        run_subprocess([sys.executable, "onnxruntime_test_python_tvm.py"], cwd=cwd, dll_path=dll_path)
+        python_path = str((Path(build_dir) / config / "_deps" / "tvm-src" / "python").resolve())
```
Comment (xadupre) [inline tools/ci_build/build.py:1979]: I don't think Path is imported. Why not using os.path.join, it is used in other places of the same file?

Comment (KJlaccHoeUM9l) [reply tools/ci_build/build.py]: @xadupre, done

**Review by xadupre — APPROVED** [review]

**Pull request merged**